fastly/service_crud_attribute_definition.go

```diff
@@ -0,0 +1,131 @@
+package fastly
+
+import (
+	"context"
+	"fmt"
+
+	gofastly "github.com/fastly/go-fastly/v3/fastly"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// ServiceCRUDAttributeDefinition is an interface for most ServiceAttributeDefinition implementations which can be
+// represented by the four CRUD operations. Most service attributes will fall into this category and should implement
+// this interface instead of ServiceAttributeDefinition directly. An attribute that implements
+// ServiceCRUDAttributeDefinition can use the ToServiceAttributeDefinition function defined below in its constructor to
+// convert it to the ServiceAttributeDefinition that the service resources expect.
+// The requirements for a service attribute to be defined in terms of this interface are:
+// - the attribute must be a nested block with a schema type of schema.TypeSet
+// - the nested block must have its own "name" attribute which uniquely defines the nested resource
+// - the block must support all four of the CRUD operations, or at least Create, Read, and Delete if updating the
+//   resource is not supported
+// Some service attributes don't fit into these constraints are better suited to implementing the
+// ServiceAttributeDefinition directly. One example is the "package" block in block_fastly_service_v1_package.go, which
+// only uses an Update operation, and therefore implements ServiceAttributeDefinition directly without
+// ServiceCRUDAttributeDefinition.
```

Comment:
![](https://media.giphy.com/media/MELUZlaowKFWF44sM1/giphy.gif)

...can we have some line breaks around this section so we have some breathing space. Otherwise it's a pretty dense wall of text 🙂 

Maybe something like (yeah I'm an 80 column width kinda guy)...

```
// ServiceCRUDAttributeDefinition is an interface for most 
// ServiceAttributeDefinition implementations which can be represented by the 
// four CRUD operations. Most service attributes will fall into this category 
// and should implement this interface instead of ServiceAttributeDefinition 
// directly. 
//
// An attribute that implements ServiceCRUDAttributeDefinition can use the 
// ToServiceAttributeDefinition function defined below in its constructor to 
// convert it to the ServiceAttributeDefinition that the service resources 
// expect.
//
// The requirements for a service attribute to be defined in terms of this 
// interface are:
//
// - the attribute must be a nested block with a schema type of schema.TypeSet
// - the nested block must have its own "name" attribute which uniquely defines 
//   the nested resource
// - the block must support all four of the CRUD operations, or at least Create, 
//   Read, and Delete if updating the resource is not supported
//
// Some service attributes don't fit into these constraints are better suited to 
// implementing the ServiceAttributeDefinition directly. One example is the 
// "package" block in block_fastly_service_v1_package.go, which only uses an 
// Update operation, and therefore implements ServiceAttributeDefinition 
// directly without ServiceCRUDAttributeDefinition.
```

Reply:
Great point, I've updated 👍🏼 I stuck it as 120 cols as I think that's what the rest of the codebase seems to be but let me know if you want me to cut it down to 80!

Ps something fishy is going on with the build 🤔

Reply:
>  I stuck it as 120 cols as I think that's what the rest of the codebase seems to be

No problem, better for consistency 👍🏻 

Yeah I was going to say the builds appear to be failing. Are the tests passing locally for you then?

Reply:
Yeah it looks like it's checking out some weird commit before running the tests? I'll see if I can get to the bottom of it

Reply:
Ok it's failing on the merged result 🤦🏼

Reply:
Aha it's due to the changes in #466 getting automerged into these changes incorrectly